### PR TITLE
KT-14241 Add Intention + Inspection to remove empty class body

### PR DESCRIPTION
**Pull request by shiraji**
#KT-14241

https://youtrack.jetbrains.com/issue/KT-14241

### CODE REVIEW

File: idea/resources/inspectionDescriptions/RemoveEmptyClassBody.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<html>
+<body>
+This inspection remove an empty class body
```
Comment (yole) [inline idea/resources/inspectionDescriptions/RemoveEmptyClassBody.html]: reports and offers to remove

File: idea/resources/intentionDescriptions/RemoveEmptyClassBodyIntention/description.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<html>
+<body>
+This intention detects an empty class body.
```
Comment (yole) [inline idea/resources/intentionDescriptions/RemoveEmptyClassBodyIntention/description.html]: removes

File: idea/src/META-INF/plugin.xml [hunk 3]
```diff
@@ -1839,6 +1844,14 @@
                      language="kotlin"
     />
 
+    <localInspection implementationClass="org.jetbrains.kotlin.idea.intentions.RemoveEmptyClassBodyInspection"
+                     displayName="Replace empty class body"
+                     groupName="Kotlin"
+                     enabledByDefault="true"
+                     level="WARNING"
```
Comment (yole) [inline idea/src/META-INF/plugin.xml]: There is nothing bad about an empty class body, so this should be severity INFO, not WARNING

File: idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.codeInspection.CleanupLocalInspectionTool
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.idea.inspections.IntentionBasedInspection
+import org.jetbrains.kotlin.psi.KtClassBody
+
+class RemoveEmptyClassBodyInspection : IntentionBasedInspection<KtClassBody>(RemoveEmptyClassBodyIntention::class), CleanupLocalInspectionTool {
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt]: If you want your inspection to run as part of code cleanup, you need to specify cleanupTool="true" in its XML registration.

File: idea/resources/inspectionDescriptions/RemoveEmptyClassBody.html (hunk 1 above)
Comment (shiraji) [inline idea/resources/inspectionDescriptions/RemoveEmptyClassBody.html]: Let me fix this

File: idea/resources/intentionDescriptions/RemoveEmptyClassBodyIntention/description.html (hunk 2 above)
Comment (shiraji) [inline idea/resources/intentionDescriptions/RemoveEmptyClassBodyIntention/description.html]: Let me fix this

File: idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt (hunk 4 above)
Comment (shiraji) [inline idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt]: OK. I didn't know that. I will add it.

**Comment by shiraji** [comment]
Thanks for reviewing. I fixed them.

**Comment by yole** [comment]
Thanks!

**Pull request merged**